pkg/controller/main-controller.go

```diff
@@ -1316,13 +1323,21 @@ func (c *Controller) syncHandler(key string) (Result, error) {
 
 	// Ensure we are only provisioning users one time
 	if !tenant.Status.ProvisionedUsers && len(tenant.Spec.Users) > 0 {
-		if err := c.createUsers(ctx, tenant, tenantConfiguration); err != nil {
-			klog.V(2).Infof("Unable to create MinIO users: %v", err)
-			c.recorder.Event(tenant, corev1.EventTypeWarning, "UsersCreatedFailed", fmt.Sprintf("Users creation failed: %s", err))
+		// Only create the user(s) until minio is ready
+		if tenant.Status.HealthStatus == miniov2.HealthStatusGreen {
```

Comment:
I guess we should also use the `StatusWaitingMinIOIsHealthy` state (also when not provisioning users).

Reply:
I am sorry, I am having a hard time understanding your comment. What do you mean exactly? Could you elaborate, please? Thank you, @ramondeklein. 🙏

Reply:
When I look at the code change, you're only using the `StatusWiatingMinIOIsHealthy` state when users need to be created. I think it would be good to always wait for the cluster to be ready before moving on to the next state. So even when no users need to be created, the `StatusWiatingMinIOIsHealthy` should be active until the cluster actually is up.

Reply:
got it, make sense, will do that